.github/workflows/build.yml

```diff
@@ -7,15 +7,23 @@ on:
       - '**.rs'
       - 'Cargo.*'
       - '*/Cargo.*'
+      - '.github/workflows/*'
```

Comment:
we should not run _this_ workflow incase the workflow files are edited, or new workflow is added.

I'm assuming it was added here so that we can trigger the workflow in this PR for testing, so it can be removed.

Reply:
> we should not run this workflow incase the workflow files are edited, or new workflow is added.

Care to explain the logic behind this?

Reply:
for example, addition of dependabot workflow, when ( if ) we would add it / change it, this workflow will get triggered, which won't add any value.

that was logic behind it.

Reply:
> which won't add any value

Certainly, GitHub does not impose fees for workflow runs on public repositories. Could you kindly elaborate on the specific value proposition you are referencing?

Reply:
value as in no point / benefit in running it.

even when Github doesn't impose fees, it feels off just to spam it. ( plus soon enough we won't be so sure why this line is required and someone gonna open PR to remove unnecessary condition haha :rofl: )

If we setup condition that workflow _must_ pass before merging, this can conflict with while adding workflows if it fails.

anyways, considering the frequency `.github/workflows/*` will be edited, I agree with you. let's keep it